### PR TITLE
feat(notched-outline): Convert JS to TypeScript

### PR DESCRIPTION
**Pull request by acdvorak**
Refs #4225

### CODE REVIEW

File: packages/mdc-notched-outline/index.ts [hunk 1]
```diff
@@ -22,35 +22,21 @@
  */
 
 import {MDCComponent} from '@material/base/component';
-
-import MDCNotchedOutlineAdapter from './adapter';
-import MDCNotchedOutlineFoundation from './foundation';
+import {MDCNotchedOutlineFoundation} from './foundation';
 import {MDCFloatingLabelFoundation} from '@material/floating-label/index';
```
Comment (acdvorak) [inline packages/mdc-notched-outline/index.ts]: Depends on #4374

File: packages/mdc-notched-outline/foundation.ts [hunk 2]
```diff
@@ -22,20 +22,14 @@
...
-  /** @return enum {string} */
+class MDCNotchedOutlineFoundation extends MDCFoundation<MDCNotchedOutlineAdapter> {
   static get strings() {
     return strings;
   }
 
-  /** @return enum {string} */
   static get cssClasses() {
     return cssClasses;
   }
```
Comment (moog16) [inline packages/mdc-notched-outline/foundation.ts]: remove the @return enum {number} below

File: packages/mdc-notched-outline/index.ts [hunk 3]
```diff
@@ -63,10 +49,10 @@ class MDCNotchedOutline extends MDCComponent {
...
 
   /**
-    * Updates classes and styles to open the notch to the specified width.
-    * @param {number} notchWidth The notch width in the outline.
-    */
-  notch(notchWidth) {
+   * Updates classes and styles to open the notch to the specified width.
+   * @param notchWidth The notch width in the outline.
+   */
+  notch(notchWidth: number): void {
```
Comment (moog16) [inline packages/mdc-notched-outline/index.ts]: remove void

**Comment by moog16** [comment]
also update `js-bundle-factory`

File: packages/mdc-notched-outline/foundation.ts (hunk 2 above)
Comment (acdvorak) [reply packages/mdc-notched-outline/foundation.ts]: Done

File: packages/mdc-notched-outline/index.ts (hunk 3 above)
Comment (acdvorak) [reply packages/mdc-notched-outline/index.ts]: Done

**Pull request merged**